### PR TITLE
P2 signup: add 'post-email confirmation' screen

### PR DESCRIPTION
**Pull request by annemirasol**
#### Changes proposed in this Pull Request

* Part of the revised P2 signup flow project (p9lV3a-2MW-p2). 
* Signup flow: display success screen if user has verified their email during signup.
* See p9lV3a-2Lp-p2#comment-3599 for details about this step.

#### Testing instructions

* `ENABLE_FEATURES=no-force-sympathy yarn start`

1. Check that verified users are skipped over:
    a. While logged in as a verified user, visit `calypso.localhost:3000/start/p2-new`. Since your user account is already verified, you should go straight to the `p2-site` step, skipping the `p2-confirm-email` step.
2. Check that unverified users are made to go through the email verification step:
    a. In incognito mode, visit `calypso.localhost:3000/start/p2-new` and sign up for a new acccount, using an email address that you have access to.
    b. After signing up, you should get the "Check your email" step. Do not close this window.
    c. Copy the verification link and paste it in a browser tab, within the same incognito session. This should mark your test user account as verified, but will redirect to Calypso home. The redirect will be fixed in a separate PR.
    d. Return to the window in 2b and reload the page. You should get the "Email confirmed" notice. Click the "Continue" button and you should get the next step.

#### Screens
Screen Shot 2021-12-28 at 4 10 27 PM

### CODE REVIEW

File: client/signup/steps/p2-email-confirmed/index.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		goToNextStep();
+	};
+
+	return (
+		<P2StepWrapper
+			flowName={ flowName }
+			stepName={ stepName }
+			positionInFlow={ positionInFlow }
+			headerIcon={ check }
+			headerText={ translate( 'Email confirmed' ) }
```
Comment (a8ci18n) [inline client/signup/steps/p2-email-confirmed/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **28** times:
`translate( 'Email confirmed!' )` *ES Score: 10*

File: client/signup/steps/p2-confirm-email/index.jsx [hunk 2]
```diff
@@ -75,6 +92,43 @@ function P2ConfirmEmail( { flowName, stepName, positionInFlow } ) {
...
+	};
+
+	return (
+		<P2StepWrapper
+			flowName={ flowName }
+			stepName={ stepName }
+			positionInFlow={ positionInFlow }
+			headerIcon={ isEmailVerified ? check : mailIcon }
+			headerText={
+				isEmailVerified ? translate( 'Email confirmed' ) : translate( 'Check your email' )
```
Comment (a8ci18n) [inline client/signup/steps/p2-confirm-email/index.jsx:132]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **28** times:
`translate( 'Email confirmed!' )` *ES Score: 10*

**Review by unDemian — CHANGES_REQUESTED** [review]
> Not sure if I'm missing something, since I've tried it twice, but I can only verify the account if I login first in the new tab and only afterwards paste the activation link. 

Hmm, that's weird. Sounds like the second tab doesn't share your logged in session with the signup flow tab? After creating an acccount, you should be automatically logged in as that account, so it's surprising that it prompts you to log in.

When you go to wordpress.com in the second tab, does it show you are logged out?

> However the issue is that reloading the "Check your email" page after I activate the account remains the same. If in the other tab I logout or I don't login in the first place reloading "Check your email" redirects me to http://calypso.localhost:3000/start/p2-new/user

`http://calypso.localhost:3000/start/p2-new/user` is what a logged out user would get. If you had manually logged out via the second tab, this makes sense.

When you activated your account, did it redirect you to `wordpress.com/read?verified=1` and showed a notice like this?
Screen Shot 2021-12-28 at 11 45 32 PM

Also, sorry I forgot to put that you will need to build with `ENABLE_FEATURES=no-force-sympathy yarn start`. Not doing this shouldn't have caused being stuck on the "Check your email" page, but it may cause skipping the "Email confirmed" page when you reload post-activation, since state rehydration might be skipped.

**Comment by annemirasol** [comment]
Here's a recording of what I'm seeing, if it helps: https://d.pr/v/sqIQQG

**Comment by unDemian** [comment]
Not sure what is happening, https://d.pr/v/yFH1oo but I still cannot reproduce it 🙁

**Comment by annemirasol** [comment]
> Not sure what is happening, https://d.pr/v/yFH1oo but I still cannot reproduce it 🙁

Ahh I see now. Please turn off the Chrome incognito's third-party cookies blocking, as there'd be trouble due to `calypso.localhost:3000` vs `wordpress.com` cookies. TIL logging in wouldn't work from calypso.localhost:3000 if third-party cookies are blocked. 


Screen Shot 2021-12-29 at 4 40 51 PM

**Review by unDemian — APPROVED** [review]
Looking good!

> Ahh I see now. Please turn off the Chrome incognito's third-party cookies blocking, as there'd be trouble due to calypso.localhost:3000 vs wordpress.com cookies. 

Great catch.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7066240

Thank you @annemirasol for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7066240).

**Pull request merged**